indexing-service/src/main/java/org/apache/druid/indexing/overlord/LockRequestForNewSegment.java

```diff
@@ -0,0 +1,183 @@
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.druid.indexing.common.LockGranularity;
+import org.apache.druid.indexing.common.TaskLock;
+import org.apache.druid.indexing.common.TaskLockType;
+import org.apache.druid.indexing.common.task.Task;
+import org.apache.druid.java.util.common.DateTimes;
+import org.apache.druid.timeline.partition.ShardSpecFactory;
+import org.joda.time.Interval;
+
+import javax.annotation.Nullable;
+
+public class LockRequestForNewSegment implements LockRequest
+{
+  private final TaskLockType lockType;
+  private final String groupId;
+  private final String dataSource;
+  private final Interval interval;
+  private final ShardSpecFactory shardSpecFactory;
+  private final int priority;
+  private final String sequenceName;
+  @Nullable
+  private final String previsousSegmentId;
+  private final boolean skipSegmentLineageCheck;
+
+  public LockRequestForNewSegment(
+      TaskLockType lockType,
+      String groupId,
+      String dataSource,
+      Interval interval,
+      ShardSpecFactory shardSpecFactory,
+      int priority,
+      String sequenceName,
+      @Nullable String previsousSegmentId,
+      boolean skipSegmentLineageCheck
+  )
+  {
+    this.lockType = lockType;
+    this.groupId = groupId;
+    this.dataSource = dataSource;
+    this.interval = interval;
+    this.shardSpecFactory = shardSpecFactory;
+    this.priority = priority;
+    this.sequenceName = sequenceName;
+    this.previsousSegmentId = previsousSegmentId;
+    this.skipSegmentLineageCheck = skipSegmentLineageCheck;
+  }
+
+  @VisibleForTesting
+  public LockRequestForNewSegment(
+      TaskLockType lockType,
+      Task task,
+      Interval interval,
+      ShardSpecFactory shardSpecFactory,
+      String sequenceName,
+      @Nullable String previsousSegmentId,
+      boolean skipSegmentLineageCheck
+  )
+  {
+    this(
+        lockType,
+        task.getGroupId(),
+        task.getDataSource(),
+        interval,
+        shardSpecFactory,
+        task.getPriority(),
+        sequenceName,
+        previsousSegmentId,
+        skipSegmentLineageCheck
+    );
+  }
+
+  @Override
+  public LockGranularity getGranularity()
+  {
+    return LockGranularity.SEGMENT;
+  }
+
+  @Override
+  public TaskLockType getType()
+  {
+    return lockType;
+  }
+
+  @Override
+  public String getGroupId()
+  {
+    return groupId;
+  }
+
+  @Override
+  public String getDataSource()
+  {
+    return dataSource;
+  }
+
+  @Override
+  public Interval getInterval()
+  {
+    return interval;
+  }
+
+  @Override
+  public int getPriority()
+  {
+    return priority;
+  }
+
+  public ShardSpecFactory getShardSpecFactory()
+  {
+    return shardSpecFactory;
+  }
+
+  @Override
+  public String getVersion()
+  {
+    return DateTimes.nowUtc().toString();
+  }
+
+  @Override
+  public boolean isRevoked()
+  {
+    return false;
+  }
+
+  @Override
+  public TaskLock toLock()
+  {
+    throw new UnsupportedOperationException();
```

Comment:
Can you add a comment here about how `LockRequestForNewSegment` gets converted to `SpecificSegmentLockRequest` in `TaskLockbox`?

Reply:
Added an exception message.